modules/git/repo_commit.go

```diff
@@ -220,6 +225,95 @@ type CommitsByFileAndRangeOptions struct {
 	Page     int
 }
 
+// ExtendedCommitStats return the list of *api.ExtendedCommitStats for the given revision
+func (repo *Repository) ExtendedCommitStats(revision string /*, limit int */) ([]*api.ExtendedCommitStats, error) {
+	baseCommit, _ := repo.GetCommit(revision)
+	stdoutReader, stdoutWriter, err := os.Pipe()
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		_ = stdoutReader.Close()
+		_ = stdoutWriter.Close()
+	}()
+
+	gitCmd := NewCommand(repo.Ctx, "log", "--shortstat", "--no-merges", "--pretty=format:---%n%cN%n%cE%n%cs", "--reverse")
+	// AddOptionFormat("--max-count=%d", limit)
+	gitCmd.AddDynamicArguments(baseCommit.ID.String())
+
+	var extendedCommitStats []*api.ExtendedCommitStats
+	stderr := new(strings.Builder)
+	err = gitCmd.Run(&RunOpts{
+		Dir:    repo.Path,
+		Stdout: stdoutWriter,
+		Stderr: stderr,
+		PipelineFunc: func(ctx context.Context, cancel context.CancelFunc) error {
+			_ = stdoutWriter.Close()
+			scanner := bufio.NewScanner(stdoutReader)
+			scanner.Split(bufio.ScanLines)
+
+			for scanner.Scan() {
+				line := strings.TrimSpace(scanner.Text())
+				if line != "---" {
+					continue
+				}
+				scanner.Scan()
+				authorName := strings.TrimSpace(scanner.Text())
+				scanner.Scan()
+				authorEmail := strings.TrimSpace(scanner.Text())
+				scanner.Scan()
+				date := strings.TrimSpace(scanner.Text())
+				scanner.Scan()
+				stats := strings.TrimSpace(scanner.Text())
+				if authorName == "" || authorEmail == "" || date == "" || stats == "" {
+					// FIXME: find a better way to parse the output so that we will handle this properly
+					log.Warn("Something is wrong with git log output, skipping...")
+					log.Warn("authorName: %s,  authorEmail: %s,  date: %s,  stats: %s", authorName, authorEmail, date, stats)
+					continue
+				}
+				//  1 file changed, 1 insertion(+), 1 deletion(-)
+				fields := strings.Split(stats, ",")
```

Comment:
example data:
```
 73930	---
 73931	GitHub
 73932	noreply@github.com
 73933	2023-07-06
 73934	 30 files changed, 461 insertions(+), 399 deletions(-)
       
 73935	---
 73936	GitHub
 73937	noreply@github.com
 73938	2023-07-06
 73939	 1 file changed, 3 insertions(+), 14 deletions(-)
       
 73940	---
 73941	GitHub
 73942	noreply@github.com
 73943	2023-07-06
 73944	 2 files changed, 57 deletions(-)
       
 73945	---
 73946	GiteaBot
 73947	teabot@gitea.io
 73948	2023-07-07
 73949	 1 file changed, 96 insertions(+)
       
 73950	---
 73951	GitHub
 73952	noreply@github.com
 73953	2023-07-07
 73954	 32 files changed, 254 insertions(+), 239 deletions(-)
```
I think we can move the parsing of this into it's own non-exported-function  and add some unit tests to that :)

Other option would to write tests for all cases for the whole func.
In any case, we should add at least a simple test for ExtendedCommitStats

If you need help, tell me:)

Reply:
also this func as the problem, that the commiter not author is used ...

e.g. 6375419468edc95fdfac94aac3b0e10b23743557
-> Commiter: `GitHub <noreply@github.com>`
-> Author: `Lunny Xiao <xiaolunwen@gmail.com>`

Reply:
>I think we can move the parsing of this into it's own non-exported-function and add some unit tests to that :)

@6543 Yeah, I agree. This is also my least favorite part of my implementation. I think splitting the parsing logic to its own function and even writing some unit tests would be great. Can you possibly help me with this part? There should be a more elegant way to parse this other than putting `scanner.Scan(), scanner.Text()` all over the place right?

>also this func as the problem, that the commiter not author is used ...

Fixed it with 056745b

Reply:
sure :)